File: concert-ticketing/src/main/java/com/hhplus/concertticketing/common/exception/CustomException.java
package com.hhplus.concertticketing.common.exception;

import lombok.Getter;

@Getter
public class CustomException extends RuntimeException {
    private final ErrorCode errorCode;

    public CustomException(ErrorCode errorCode) {
        super(errorCode.getCode());
        this.errorCode = errorCode;
    }
    public CustomException(ErrorCode errorCode, String message) {
        super(message);
        this.errorCode = errorCode;
    }
}

File: concert-ticketing/src/main/java/com/hhplus/concertticketing/common/exception/CustomException.kt
package com.hhplus.concertticketing.common.exception

/**
 * 서비스 전반에서 사용되는 커스텀 예외
 */
class CustomException(
    val errorCode: ErrorCode,
    message: String? = null
) : RuntimeException(message ?: errorCode.code)
